flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolUtils.java

```diff
@@ -45,19 +50,108 @@
  */
 public class SlotPoolUtils {
 
+	public static final Time TIMEOUT = Time.seconds(10L);
+
 	private SlotPoolUtils() {
 		throw new UnsupportedOperationException("Cannot instantiate this class.");
 	}
 
+	public static SlotPoolBuilder createSlotPoolBuilder() {
+		return createSlotPoolBuilder(ComponentMainThreadExecutorServiceAdapter.forMainThread());
+	}
+
+	public static SlotPoolBuilder createSlotPoolBuilder(final ComponentMainThreadExecutor mainThreadExecutor) {
+		return new SlotPoolBuilder(mainThreadExecutor);
+	}
+
+	public static SlotPoolBuilder createSlotPoolBuilder(final ResourceManagerGateway resourceManagerGateway) {
+		return createSlotPoolBuilder().setResourceManagerGateway(resourceManagerGateway);
+	}
+
+	public static SlotPoolBuilder createSlotPoolBuilder(
+			final ComponentMainThreadExecutor mainThreadExecutor,
+			final ResourceManagerGateway resourceManagerGateway) {
+
+		return createSlotPoolBuilder(mainThreadExecutor).setResourceManagerGateway(resourceManagerGateway);
+	}
+
+	public static TestingSlotPoolImpl createAndSetUpSlotPool(
+			final ResourceManagerGateway resourceManagerGateway) throws Exception {
+
+		return createSlotPoolBuilder(resourceManagerGateway).build();
+	}
+
+	public static TestingSlotPoolImpl createAndSetUpSlotPool(
+			final ResourceManagerGateway resourceManagerGateway,
+			final JobID jobId) throws Exception {
+
+		return createSlotPoolBuilder(resourceManagerGateway).setJobId(jobId).build();
+	}
+
+	public static TestingSlotPoolImpl createAndSetUpSlotPool(
```

Comment:
This method is used only in `SlotPoolImplTest`. I'd prefer to put it there at the moment.

Reply:
Moved.